### PR TITLE
Update proxy update commit messages with tag info

### PR DESCRIPTION
**Pull request by olix0r**
Each proxy release tag now includes a message.

This change updates the git-commit-proxy-version script to include this
message in the commit message in this repo.

### CODE REVIEW

**Comment by olix0r** [comment]
Creates messages like the following:

```
proxy: v2.76.0
  
This release does not impact any user-facing behavior. The proxy's
internals have been restructured.

---

* Update maintainers.md (linkerd/linkerd2-proxy#366)
* internal: Split Accept from Listen (linkerd/linkerd2-proxy#368)
* internal: Remove the `ClientUsedTls` response extension (linkerd/linkerd2-proxy#370)
* internal: Split transport utilities into a subcrate (linkerd/linkerd2-proxy#369)
* Audit Cargo.lock for unused and outdated crates (linkerd/linkerd2-proxy#373)
* Adopt tokio-tracing contexts (linkerd/linkerd2-proxy#374)
* internal: Spilt app from linkerd2-proxy (linkerd/linkerd2-proxy#375)
* ci: Run docker builds on linkerd-docker host (linkerd/linkerd2-proxy#376)
* Update to Rust 1.38.0 (linkerd/linkerd2-proxy#377)
```

File: bin/git-commit-proxy-version [hunk 1]
```diff
@@ -37,18 +37,27 @@ if ! git rev-parse --verify --quiet "${old_proxy_rev}" ; then
...
-git log --pretty=oneline --abbrev-commit --reverse "${old_proxy_rev}..${new_proxy_rev}" | \
-    sed -E -e 's,^[a-f0-9]{7} ,* ,' \
-    -e 's, \((#[0-9]+)\), (linkerd/linkerd2-proxy\1),' \
-    >>"$tmp/commit.template"
+# Commit message
+(
+    echo "proxy: ${new_proxy_version}"
+    echo ""
+    # Include the log message, stripping the version header and GPG trailer.
+    git tag -l --format='%(contents)' release/v2.76.0 | \
```
Comment (adleong) [inline bin/git-commit-proxy-version]: should this be `$new_proxy_rev`?

Comment (adleong) [reply bin/git-commit-proxy-version]: or `$new_proxy_version`, I think...

Comment (olix0r) [reply bin/git-commit-proxy-version]: 🙏

**Pull request merged**